src/ReaderScreen/hooks/useFetchShabad.js

```diff
@@ -13,25 +13,34 @@ const useFetchShabad = (shabadID) => {
   const isLarivaarAssist = useSelector((state) => state.isLarivaarAssist);
   const isParagraphMode = useSelector((state) => state.isParagraphMode);
   const isVishraam = useSelector((state) => state.isVishraam);
+  // const prevBaniLengthRef = useRef(baniLength);
+  // const prevTransliterationLanguageRef = useRef(transliterationLanguage);
+  // const prevVishraamSourceRef = useRef(vishraamSource);
+  // const prevVishraamOptionRef = useRef(vishraamOption);
+  // const prevIsLarivaarRef = useRef(isLarivaar);
+  // const prevIsLarivaarAssistRef = useRef(isLarivaarAssist);
+  // const prevIsParagraphaMode = useRef(isParagraphMode);
+  // const prevIsVishraam = useRef(isVishraam);
+
+  const fetchShabad = async () => {
```

Comment:
Similar to above, we can add the `useCallback` to memorize the function definition with the state sideEffects.

```javascript
const fetchShabad = useCallback(async() => {
     // function code here


}, [state1, state2...]);

```